app/code/Magento/InventoryConfiguration/Setup/Operation/CreateSourceConfigurationTable.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryConfiguration\Setup\Operation;
+
+use Magento\Framework\DB\Ddl\Table;
+use Magento\Framework\Setup\SchemaSetupInterface;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Inventory\Model\ResourceModel\SourceItem as SourceItemResourceModel;
+use Magento\InventoryConfiguration\Model\ResourceModel\SourceItemConfiguration;
+use Magento\InventoryConfiguration\Api\Data\SourceItemConfigurationInterface;
+
+class CreateSourceConfigurationTable
+{
+    /**C
+     * @param SchemaSetupInterface $setup
+     * @return void
+     */
+    public function execute(SchemaSetupInterface $setup)
+    {
+        $notifyQtyTable = $setup->getConnection()->newTable(
+            $setup->getTable(SourceItemConfiguration::TABLE_NAME_SOURCE_ITEM_CONFIGURATION)
+        )->setComment(
+            'Inventory Notification Table'
+        );
+
+        $sourceItemTable = $setup->getTable(SourceItemResourceModel::TABLE_NAME_SOURCE_ITEM);
+
+        $notifyQtyTable = $this->addBaseFields($notifyQtyTable);
+        $notifyQtyTable = $this->addForeignKey($notifyQtyTable, $sourceItemTable, $setup);
+
+        $setup->getConnection()->createTable($notifyQtyTable);
+    }
+
+    /**
+     * Add columns to table.
+     *
+     * @param Table $notifyQtyTable
+     * @return Table
+     */
+    private function addBaseFields(Table $notifyQtyTable): Table
+    {
+        return $notifyQtyTable->addColumn(
+            SourceItemConfigurationInterface::SOURCE_ITEM_ID,
+            Table::TYPE_INTEGER,
+            null,
+            [
+                Table::OPTION_IDENTITY => true,
+                Table::OPTION_UNSIGNED => true,
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_PRIMARY => true,
+            ],
+            'Source Item ID'
+        )->addColumn(
+            SourceItemConfigurationInterface::INVENTORY_NOTIFY_QTY,
+            Table::TYPE_DECIMAL,
+            null,
+            [
+                Table::OPTION_UNSIGNED => false,
+                Table::OPTION_NULLABLE => false,
```

Comment:
Current field declaration is
`notify_stock_qty` DECIMAL(12,4) NULL DEFAULT NULL COMMENT 'Notify Stock Qty',

We need to have the same declaration for Backward Compatibility
Pay attention that this fields could be null